utilityFunctions.js

```diff
@@ -0,0 +1,104 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const { exit } = require('process');
+const app = express();
+
+app.use(bodyParser.json());
+
+// I declare these lines so they are global here so i dont have to declare them in every function.
+// The append part does not work anymore because when reading the file user.json when not available the app crashes, i dont have a solution for this yet.
+
+const fs = require('fs');
+const fileData = JSON.parse(fs.readFileSync('user.json'));
+const path = './user.json';
+
+const validationUserPost = (userPostData) => {
+    const postedName = userPostData.name;
+    const postedHair = userPostData.hair;
+    const postedHeight = userPostData.height;
+    var regEx = /^[A-Za-z]+$/;
+    if (postedName.match(regEx) && postedHair.match(regEx) &&
+        postedName.length >= 4 && postedName.length <= 50 && postedHeight > 50 && postedHeight < 250) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+const addUserPost = (userPostData) => {
+    if (fs.existsSync(path) != true) {
+        fs.appendFileSync('user.json', JSON.stringify([userPostData], null, 2));
+    } else {
+        fileData.push(userPostData)
+        fs.writeFileSync('user.json', JSON.stringify(fileData, null, 2));
+    }
+    res.status(201).json({
+        message: `Successfully created the user ${userPostData.name}`,
+    })
+}
+
+const putUserbyId = (userPutData, parameterId) => {
+    if (fs.existsSync(path) != true) {
+        fs.appendFileSync('user.json', JSON.stringify([userPutData], null, 2));
+    }
+    let indexPut = fileData.findIndex(updatedObject => {
+        return updatedObject.name === `${parameterId}`;
+    });
+    if (indexPut > -1) {
+        fileData.splice(indexPut, 1, userPutData);
+        fs.writeFileSync('user.json', JSON.stringify(fileData, null, 2));
+        return true;
+    } else {
+        return false;
+    }
+}
+
+const deleteUserById = (parameterId) => {
+    const tobedeletedObject = parameterId;
+    let indexDel = fileData.findIndex(tobedeletedObject => {
+        return tobedeletedObject.name === `${parameterId}`;
+    });
+    if (indexDel > -1) {
+        fileData.splice(indexDel, 1);
+        fs.writeFileSync('user.json', JSON.stringify(fileData, null, 2));
+        return true;
+    } else if (indexDel < 0) {
+        return false;
+    }
+}
+
+const getUserById = (requestedId) => {
+    const requestedObject = requestedId;
+    let indexGet = fileData.findIndex(requestedObject => {
+        return requestedObject.name === `${requestedId}`;
+    })
+    const responseUserFile = fileData[indexGet];
+    if (indexGet > -1) {
+        return responseUserFile;
+    }
+    else {
+        return false;
+
+    }
+};
+
+const readFromUserFile = () => {
+    return fileData;
```

Comment:
We have a bug here because you are always using a value of this variable that was created during initialization. Can you see the consequences? If not, we can skip it for now because probably you will see it on our next assessment ;)

Reply:
Yes I assume you refer to fileData which is indeed on one of the first lines. I understand that this should be a bug because of the fact that the data changes during some requests etc and I use this at the end of the file. But when I request the get after some updates/posts then the data is still good. So I don't no why it still works and doesn't look like a problem now. But for later is also fine of course:)